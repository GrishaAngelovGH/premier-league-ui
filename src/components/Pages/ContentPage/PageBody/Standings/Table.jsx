import PropTypes from 'prop-types'
import { statuses } from 'images'

const Table = ({ rows, columns }) => (
    <table className='table table-hover table-sm table-responsive-sm'>
        <thead className='text-secondary'>
            <tr>
                {
                    columns.map(v => (
                        <th key={v} scope='col'>{v}</th>
                    ))
                }
            </tr>
        </thead>
        <tbody>
            {
                rows.map(v => (
                    <tr key={v.position}>
                        <th>
                            <span className='pr-2'>{v.position}</span>
                            <img src={v.image} width={25} />
                            <span className='pl-2'>{v.title}</span>
                        </th>
                        <td>{v.mp}</td>
                        <td>{v.w}</td>
                        <td>{v.d}</td>
                        <td>{v.l}</td>
                        <td>{v.gf}</td>
                        <td>{v.ga}</td>
                        <td>{v.gd}</td>
                        <td>{v.pts}</td>
                        <td>
                            {
                                v.lastFive.map((v, i) => {
                                    if (v === 'w') return <img key={i} src={statuses['win']} />
                                    if (v === 'd') return <img key={i} src={statuses['draw']} />
                                    if (v === 'l') return <img key={i} src={statuses['loss']} />
                                })
                            }
                        </td>
                    </tr>
                ))
            }
        </tbody>
    </table>
)

Table.propTypes = {
    rows: PropTypes.array.isRequired,
    columns: PropTypes.array.isRequired
}

export default Table