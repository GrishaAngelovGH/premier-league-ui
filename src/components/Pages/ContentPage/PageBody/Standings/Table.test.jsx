import { render } from '@testing-library/react'

import Table from './Table'

test('should render Table component', () => {
	const view = render(
		<Table
			rows={[
				{
					position: 1,
					image: 'tottenham-image',
					title: 'Tottenham',
					mp: 10,
					w: 6,
					d: 3,
					l: 1,
					gf: 21,
					ga: 9,
					gd: 12,
					pts: 21,
					lastFive: ['d', 'w', 'w', 'w', 'w']
				}
			]}
			columns={['Club', 'MP', 'W', 'D', 'L', 'GF', 'GA', 'GD', 'Pts', 'Last 5']}
		/>
	)

	expect(view).toMatchSnapshot()
})