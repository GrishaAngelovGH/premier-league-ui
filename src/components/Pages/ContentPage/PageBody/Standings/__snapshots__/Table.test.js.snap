// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table table-hover table-sm table-responsive-sm"
      >
        <thead
          class="text-secondary"
        >
          <tr>
            <th
              scope="col"
            >
              Club
            </th>
            <th
              scope="col"
            >
              MP
            </th>
            <th
              scope="col"
            >
              W
            </th>
            <th
              scope="col"
            >
              D
            </th>
            <th
              scope="col"
            >
              L
            </th>
            <th
              scope="col"
            >
              GF
            </th>
            <th
              scope="col"
            >
              GA
            </th>
            <th
              scope="col"
            >
              GD
            </th>
            <th
              scope="col"
            >
              Pts
            </th>
            <th
              scope="col"
            >
              Last 5
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              <span
                class="pr-2"
              >
                1
              </span>
              <img
                src="tottenham-image"
                width="25"
              />
              <span
                class="pl-2"
              >
                Tottenham
              </span>
            </th>
            <td>
              10
            </td>
            <td>
              6
            </td>
            <td>
              3
            </td>
            <td>
              1
            </td>
            <td>
              21
            </td>
            <td>
              9
            </td>
            <td>
              12
            </td>
            <td>
              21
            </td>
            <td>
              <img
                src="draw.svg"
              />
              <img
                src="win.svg"
              />
              <img
                src="win.svg"
              />
              <img
                src="win.svg"
              />
              <img
                src="win.svg"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="table table-hover table-sm table-responsive-sm"
    >
      <thead
        class="text-secondary"
      >
        <tr>
          <th
            scope="col"
          >
            Club
          </th>
          <th
            scope="col"
          >
            MP
          </th>
          <th
            scope="col"
          >
            W
          </th>
          <th
            scope="col"
          >
            D
          </th>
          <th
            scope="col"
          >
            L
          </th>
          <th
            scope="col"
          >
            GF
          </th>
          <th
            scope="col"
          >
            GA
          </th>
          <th
            scope="col"
          >
            GD
          </th>
          <th
            scope="col"
          >
            Pts
          </th>
          <th
            scope="col"
          >
            Last 5
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th>
            <span
              class="pr-2"
            >
              1
            </span>
            <img
              src="tottenham-image"
              width="25"
            />
            <span
              class="pl-2"
            >
              Tottenham
            </span>
          </th>
          <td>
            10
          </td>
          <td>
            6
          </td>
          <td>
            3
          </td>
          <td>
            1
          </td>
          <td>
            21
          </td>
          <td>
            9
          </td>
          <td>
            12
          </td>
          <td>
            21
          </td>
          <td>
            <img
              src="draw.svg"
            />
            <img
              src="win.svg"
            />
            <img
              src="win.svg"
            />
            <img
              src="win.svg"
            />
            <img
              src="win.svg"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
